agent/src/main/resources/build-scan-init.gradle

```diff
@@ -156,6 +167,14 @@ if (GradleVersion.current() < GradleVersion.version('6.0')) {
                     ext.buildScan.value 'CI auto injection', 'TeamCity'
                 }
             }
+
+            if (geUrl && geOverrideServerUrl) {
```

Comment:
It is very tempting to try to combine this logic with the logic in the `if (!settings.pluginManager...)` block. I did not do so for this PR. But we could consider the following (pseudocode):

```
if plugin is not applied:
	apply plugin
    add auto injection value

if plugin was applied or if ge url is set and override is enabled:
	set server url
    set allow untrusted
    set publish always
    disable upload in backgroun
```